### PR TITLE
Feature/email verification

### PR DESCRIPTION
**Pull request by hanaum**

### CODE REVIEW

File: js/src/http/UserDirectoryApi.coffee [hunk 1]
```diff
@@ -22,6 +22,8 @@ define 'kryptnostic.user-directory-api', [
   getUsersUrl  = -> Configuration.get('heraclesUrl') + '/directory/users'
   usersInRealmUrl = -> Configuration.get('heraclesUrl') + '/directory'
   setFirstLoginUrl = -> Configuration.get('heraclesUrl') + '/directory/setlogin'
+  getConfirmation = -> Configuration.get('heraclesUrl') + '/registration/confirmation/resend'
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: the pattern for these ends in `Url`, so:

`getConfirmationUrl`
`getVerificationUrl`

Comment (hanaum) [inline js/src/http/UserDirectoryApi.coffee]: fixed.

File: js/src/http/UserDirectoryApi.coffee [hunk 2]
```diff
@@ -195,4 +197,20 @@ define 'kryptnostic.user-directory-api', [
...
+        Requests.wrapCredentials({
+          url: getConfirmation()
+          method: 'GET'
+        }, credentials)
+      ))
+
+    sendConfirmationToken: (uuid, token) ->
+      Promise.resolve(axios(
+        Requests.wrapCredentials({
+          url: getVerification() + uuid + '/' + token
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: pass these into the function:

`url: getVerification(uuid, token)`

File: js/src/http/UserDirectoryApi.coffee [hunk 3]
```diff
@@ -22,6 +22,8 @@ define 'kryptnostic.user-directory-api', [
   getUsersUrl  = -> Configuration.get('heraclesUrl') + '/directory/users'
   usersInRealmUrl = -> Configuration.get('heraclesUrl') + '/directory'
   setFirstLoginUrl = -> Configuration.get('heraclesUrl') + '/directory/setlogin'
+  getConfirmation = -> Configuration.get('heraclesUrl') + '/registration/confirmation/resend'
+  getVerification = -> Configuration.get('heraclesUrl') + '/registration/verification/'
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: have this take in the params from below:

`getVerification = (uuid, token) -> Configuration.get('heraclesUrl') + '/registration/verification/' + uuid + '/' + token`

File: js/src/service/RegistrationClient.coffee [hunk 4]
```diff
@@ -46,7 +48,13 @@ define 'kryptnostic.registration-client', [
         @registrationApi.register(userRegistrationRequest)
       .then (uuid) =>
         log.info('registered new user account', { uuid })
-        @credentialService.initializeSalt({ uuid, encryptedSalt, credential })
+        Promise.all([
+          @credentialService.initializeSalt({ uuid, encryptedSalt, credential }),
+          @userDirectoryApi.resendConfirmationEmail({
+            principal  : uuid,
+            credential : credential
+            })
```
Comment (UnsungHero97) [inline js/src/service/RegistrationClient.coffee]: fix indentation

File: js/src/service/RegistrationClient.coffee [hunk 5]
```diff
@@ -46,7 +48,13 @@ define 'kryptnostic.registration-client', [
...
+          ])
```
Comment (UnsungHero97) [inline js/src/service/RegistrationClient.coffee]: fix indentation

File: js/src/service/RegistrationClient.coffee [hunk 6]
```diff
@@ -12,6 +12,7 @@ define 'kryptnostic.registration-client', [
   CredentialService       = require 'kryptnostic.credential-service'
   UserRegistrationRequest = require 'kryptnostic.user-registration-request'
   KryptnosticWorkersApi   = require 'kryptnostic.kryptnostic-workers-api'
+  UserDirectoryApi        = require 'kryptnostic.user-directory-api'
```
Comment (UnsungHero97) [inline js/src/service/RegistrationClient.coffee]: you need to add `'kryptnostic.user-directory-api'` to the list at the top of the file

File: js/src/http/UserDirectoryApi.coffee [hunk 7]
```diff
@@ -22,6 +22,8 @@ define 'kryptnostic.user-directory-api', [
   getUsersUrl  = -> Configuration.get('heraclesUrl') + '/directory/users'
   usersInRealmUrl = -> Configuration.get('heraclesUrl') + '/directory'
   setFirstLoginUrl = -> Configuration.get('heraclesUrl') + '/directory/setlogin'
+  getConfirmationUrl = -> Configuration.get('heraclesUrl') + '/registration/confirmation/resend'
+  getVerificationUrl = -> Configuration.get('heraclesUrl') + '/registration/verification/'
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: have this take in the params from below:

```
getVerification = (uuid, token) ->
  Configuration.get('heraclesUrl') + '/registration/verification/' + uuid + '/' + token
```

you can leave it as a single line

File: js/src/http/UserDirectoryApi.coffee [hunk 8]
```diff
@@ -195,4 +197,20 @@ define 'kryptnostic.user-directory-api', [
...
+        Requests.wrapCredentials({
+          url: getConfirmationUrl()
+          method: 'GET'
+        }, credentials)
+      ))
+
+    sendConfirmationToken: (uuid, token) ->
+      Promise.resolve(axios(
+        Requests.wrapCredentials({
+          url: getVerificationUrl() + uuid + '/' + token
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: pass these into the function:

`url: getVerification(uuid, token)`

**Pull request merged**